ui/src/app/base/components/TooltipButton/TooltipButton.tsx

```diff
@@ -0,0 +1,52 @@
+import type { ReactNode } from "react";
+
+import type {
+  ButtonProps,
+  IconProps,
+  SubComponentProps,
+  TooltipProps,
+} from "@canonical/react-components";
+import { Button, Icon, Tooltip } from "@canonical/react-components";
+
+import { breakLines, unindentString } from "app/utils";
+
+type Props = Omit<TooltipProps, "children"> & {
+  buttonProps?: SubComponentProps<ButtonProps>;
+  children?: ReactNode;
+  iconProps?: SubComponentProps<IconProps>;
+};
+
+const defaultButtonProps: ButtonProps = {
+  appearance: "base",
+  className: "u-display--block u-no-border u-no-margin",
+  hasIcon: true,
+  small: true,
+  type: "button",
+};
+
+const defaultIconProps: IconProps = { name: "information" };
```

Comment:
I'm curious as to why these props are here vs included in the JSX. Was there a type issue of some kind?

Reply:
I... don't have a good answer why it's like this. It just came out of my brain this way? Anyway now that you've mentioned it I'm gonna change it to handling it in the JSX, like every other instance.

🧠 🚫